### PR TITLE
refactor(display-name): rendered display name component in react

### PR DESCRIPTION
**Pull request by hritvi**
Some screenshots:
Screenshot from 2019-06-12 10-16-02

Screenshot from 2019-06-12 10-16-18

Screenshot from 2019-06-12 10-16-53

Screenshot from 2019-06-12 10-17-15

@shane-tomlinson @lmorchard Please review.

### CODE REVIEW

File: packages/fxa-content-server/app/tests/spec/views/settings/display_name.jsx [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+            return Promise.resolve();
+          });
+          sinon.stub(view, 'displaySuccess').callsFake(function () {
+            return Promise.resolve();
+          });
+          sinon.spy(view, 'logFlowEvent');
+          sinon.spy(view, 'render');
+          sinon.spy(view, 'navigate');
+
+          // view.$('input.display-name').val(name);
```
Comment (shane-tomlinson) [inline packages/fxa-content-server/app/tests/spec/views/settings/display_name.jsx]: Please remove this line.

File: packages/fxa-content-server/package.json [hunk 2]
```diff
@@ -47,6 +48,7 @@
     "body-parser": "1.18.2",
     "cache-loader": "^3.0.1",
     "celebrate": "7.0.3",
+    "chai": "^4.2.0",
```
Comment (shane-tomlinson) [inline packages/fxa-content-server/package.json]: Can you move this to devDependencies?

File: packages/fxa-content-server/app/tests/spec/views/settings/display_name.jsx [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    });
+
+    it('validates the display name field for changes:name changed',() => {
+      account.set('displayName', 'joe');
+      renderDisplayNameFormComponent('joe changed');
+
+      const submitDisplay = document.querySelectorAll('#submit_display');
+      assert.lengthOf(submitDisplay, 1);
+      assert.equal(submitDisplay[0].attributes.disabled,null,'name changed');
+    });
```
Comment (shane-tomlinson) [inline packages/fxa-content-server/app/tests/spec/views/settings/display_name.jsx]: Is there a way to test form submission and ensure the expected function is called with the expected parameters?

**Comment by hritvi** [comment]
@shane-tomlinson could you please review it now. I have rebased the branch with master branch and have added the tests as you mentioned. Thanks!!

File: packages/fxa-content-server/grunttasks/l10n-extract.js [hunk 4]
```diff
@@ -19,7 +19,7 @@ var messagesOutputPath = path.join(
   'LC_MESSAGES'
 );
 var babelCmd =
-  'npx babel --plugins=@babel/plugin-syntax-dynamic-import,dynamic-import-webpack,@babel/plugin-proposal-class-properties --presets @babel/env app/scripts --out-dir .es5';
+  'npx babel --plugins=@babel/plugin-syntax-dynamic-import,dynamic-import-webpack,@babel/plugin-proposal-class-properties --presets @babel/env app/scripts --out-dir .es5 @babel/preset-react';
```
Comment (shane-tomlinson) [inline packages/fxa-content-server/grunttasks/l10n-extract.js]: `@babel/preset-react';` needs to go right before `@babel/env`, so it looks like `@babel/preset-react,@babel/env`

**Pull request merged**